pkg/scheduler/plugins/multihierarchyelasticquota/quota_operator.go

```diff
@@ -0,0 +1,135 @@
+limitations under the License.
+*/
+
+package multihierarchyelasticquota
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	QuotaKoordinatorPrefix                         = "koordinator.quota"
+	LabelQuotaIsParent                             = QuotaKoordinatorPrefix + "/is-parent"
+	LabelQuotaParent                               = QuotaKoordinatorPrefix + "/parent"
+	LabelAllowLentResource                         = QuotaKoordinatorPrefix + "/allow-lent-resource"
+	AnnotationSharedRatio                          = QuotaKoordinatorPrefix + "/shared-ratio"
+	SystemQuotaName                  QuotaNameType = "system"
+	NeedScaleMinQuotaWhenOverRootRes               = QuotaKoordinatorPrefix + "/need-scale-minquota-when-over-rootres"
+)
+
+type QuotaNameType string
+type QuotaType string
+
+func GetParentQuotaInfo(quota *v1alpha1.ElasticQuota) (isParent bool, parentName QuotaNameType, err error) {
+	isParent, err = IsParentQuota(quota)
+	if err != nil {
+		return false, "", err
+	}
+
+	parentName, err = GetParentQuotaName(quota)
+	if err != nil {
+		return isParent, "", err
+	}
+
+	return isParent, parentName, nil
+}
+
+func GetParentQuotaName(quota *v1alpha1.ElasticQuota) (QuotaNameType, error) {
+	parentName, err := GetQuotaLabelValue(quota, LabelQuotaParent)
+	if err == nil {
+		if len(parentName) <= 0 {
+			err = fmt.Errorf("parent LabelQuotaParent invalid, value:%v", parentName)
+			return QuotaNameType(parentName), err
+		}
+	}
+
+	return QuotaNameType(parentName), nil
+}
+
+func IsParentQuota(quota *v1alpha1.ElasticQuota) (isParent bool, err error) {
+	isParent = false
+	value, err := GetQuotaLabelValue(quota, LabelQuotaIsParent)
+	if err == nil {
+		isParent, err = strconv.ParseBool(value)
+		if err != nil {
+			err = fmt.Errorf("parent IsParentKey invalid, err:%v value:%v", err, value)
+			return false, err
+		}
+	}
+
+	return isParent, nil
+}
+
+func IsAllowLentResource(quota *v1alpha1.ElasticQuota) (allowLentResource bool, err error) {
+	allowLentResource = false
+	value, err := GetQuotaLabelValue(quota, LabelAllowLentResource)
+	if err == nil {
+		allowLentResource, err = strconv.ParseBool(value)
+		if err != nil {
+			err = fmt.Errorf("AllowLentResource Key invalid, err:%v value:%v", err, value)
+			return false, err
+		}
+	}
+
+	return allowLentResource, nil
+}
+
+func GetSharedRatio(quota *v1alpha1.ElasticQuota) (corev1.ResourceList, error) {
+	value, err := GetQuotaAnnotationValue(quota, AnnotationSharedRatio)
+	if err != nil {
+		return nil, err
+	}
+	resList := corev1.ResourceList{}
+	err = json.Unmarshal([]byte(value), &resList)
+	if err != nil {
+		return nil, fmt.Errorf("sharedRatio format error:%v", err)
+	}
+	return resList, nil
+}
+
+func GetQuotaLabelValue(quota *v1alpha1.ElasticQuota, key string) (value string, err error) {
+	if val, exist := quota.Labels[key]; exist {
+		value = val
```

Comment:
why if exist not return directly, but else return directly?  besides,  key not exist is very normal, I don't think is a error, may be return bool is better